ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -48,50 +53,52 @@ public class EncryptorImpl implements Encryptor {
         "The ML encryption master key has not been initialized yet. Please retry after waiting for 10 seconds.";
     private ClusterService clusterService;
     private Client client;
-    private volatile String masterKey;
+    private final Map<String, String> tenantMasterKeys;
```

Comment:
I don't get why we have different key for every tenant. Currently this key is shared on the cluster level and crosses users. It's only for the node internal use, doesn't matter with user information. If we did so, we would also need to have different key for every user.

Reply:
Tenant isn't an user here. So every tenant will have their own separate master key which will be used to encrypt/decrypt their connector credentials. If we have only one master key, that'll be a security concern for multi-tenancy.